### PR TITLE
Add cloud-run examples and integration tests

### PR DESCRIPTION
**Pull request by q2w**

### CODE REVIEW

File: examples/.gitignore [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# OSX leaves these everywhere on SMB shares
```
Comment (bharathkkb) [inline examples/.gitignore:1]: nit: remove since we have gitignore in parent

File: examples/cloud-run/main.tf [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      containers {
+        image = "gcr.io/cloudrun/hello"
+      }
+    }
+  }
+  metadata {
+    annotations = {
+      # For valid annotation values and descriptions, see
+      # https://cloud.google.com/sdk/gcloud/reference/run/deploy#--ingress
+      "run.googleapis.com/ingress" = "internal-and-cloud-load-balancing"
```
Comment (bharathkkb) [inline examples/cloud-run/main.tf:77]: Is this required for LB to function?

File: examples/cloud-run/variables.tf [hunk 3]
```diff
@@ -15,12 +15,10 @@
...
+  type = string
 }
 
 variable "region" {
-  description = "The region where resources will be created"
-  type        = string
-  default     = "us-central1"
-}
+  type    = string
+  default = "us-central1"
```
Comment (bharathkkb) [inline examples/cloud-run/variables.tf:23]: nit: lets hard code this rather than exposing

File: metadata.yaml [hunk 4]
```diff
@@ -24,9 +24,10 @@ spec:
     source:
       repo: https://github.com/googlestaging/terraform-google-regional-lb-http.git
```
Comment (bharathkkb) [inline metadata.yaml:25]: nit: update org

File: test/integration/cloud-run/cloud_run_regional_lb_test.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+func TestCloudRunRegionalLb(t *testing.T) {
+	bpt := tft.NewTFBlueprintTest(t)
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		bpt.DefaultVerify(assert)
+
+		externalIp := bpt.GetStringOutput("external_ip")
+		test.AssertResponseStatus(t, assert, "http://"+externalIp, 200)
```
Comment (bharathkkb) [inline test/integration/cloud-run/cloud_run_regional_lb_test.go:32]: Let's wrap this in a retry

https://github.com/GoogleCloudPlatform/terraform-google-cloud-run/blob/b8f6320e92d09e1a17e1744df8d7beaf187089b2/test/integration/simple_job_exec/simple_job_exec_test.go#L41

**Review by bharathkkb — APPROVED** [review]

File: main.tf [hunk 6]
```diff
@@ -14,58 +14,299 @@
...
+
+### IPv4 ###
+resource "google_compute_forwarding_rule" "default" {
+  provider              = google-beta
+  project               = var.project_id
+  region                = var.region
+  name                  = "${var.name}-forwarding-rule-http"
+  target                = google_compute_region_target_http_proxy.default[0].self_link
+  port_range            = var.http_port
+  ip_address            = local.address
```
Comment (qz267) [inline main.tf:208]: `local vars` are not defined here, need to update `variables` as well.

File: variables.tf [hunk 7]
```diff
@@ -14,38 +14,210 @@
...
+  description = "Name for the backend service."
+  type        = string
+}
+
+variable "project_id" {
   description = "The project to deploy to, if not set the default provider project is used."
   type        = string
 }
 
+variable "region" {
```
Comment (qz267) [inline variables.tf:27]: Duplication of https://github.com/qz267/tg-regional-lb-http/pull/7/files#diff-05b5a57c136b6ff596500bcbfdcff145ef6cddea2a0e86d184d9daa9a65a288eL27

**Pull request merged**